pkg/apis/clusteroperator/types.go

```diff
@@ -47,109 +48,279 @@ type ClusterList struct {
+	Hardware ClusterHardwareSpec
 
-	// ComputeNodeGroups specify the configurations of the compute node groups
+	// Config specifies cluster-wide OpenShift configuration
+	Config ClusterConfigSpec
+
+	// MachineSets specifies the configuration of all machine sets for the cluster
+	MachineSets []ClusterMachineSet
+}
+
+// ClusterHardwareSpec specifies hardware for a cluster. The specification will
+// be specific to each cloud provider.
+type ClusterHardwareSpec struct {
 	// +optional
-	ComputeNodeGroups []ClusterComputeNodeGroup
+	// AWSConfig specifies cluster hardware configuration on AWS
+	AWS *AWSClusterSpec
+
+	// TODO: Add other cloud-specific Specs as needed
 }
 
-type ClusterStatus struct {
-	// MasterNodeGroups is the number of actual master node groups that are
-	// active for the cluster
-	MasterNodeGroups int
+// AWSClusterSpec contains cluster-wide configuration for a cluster on AWS
+type AWSClusterSpec struct {
+	// AccountSeceret refers to a secret that contains the AWS account access
+	// credentials
+	AccountSecret corev1.LocalObjectReference
+
+	// Region specifies the AWS region where the cluster will be created
+	Region string
 
-	// ComputeNodeGroups is the number of actual compute node groups that are
-	// active for the cluster
-	ComputeNodeGroups int
+	// +optional
+	// VPCName specifies the name of the VPC to associate with the cluster.
+	// If existing, it will not be created. If left blank, the VPC name will
+	// be derived from the cluster name.
+	VPCName string
+
+	// VPCSubnet specifies the subnet to use for the cluster's VPC. Only used
+	// when a new VPC is created for the cluster
+	VPCSubnet string
 }
 
-// ClusterNodeGroup is a node group defined in a Cluster resource
-type ClusterNodeGroup struct {
-	Size int
+// ClusterConfigSpec contains OpenShift configuration for a cluster
+type ClusterConfigSpec struct {
+	// DeploymentType indicates the type of OpenShift deployment to create
+	DeploymentType string
+
+	// OpenShiftVersion is the version of OpenShift to install
+	OpenshiftVersion string
+
+	// SDNPluginName is the name of the SDN plugin to use for this install
+	SDNPluginName string
+
+	// ServiceNetworkSubnet is the CIDR to use for service IPs in the cluster
+	ServiceNetworkSubnet string
+
+	// PodNetworkSubnet is the CIDR to use for pod IPs in the cluster
+	PodNetworkSubnet string
 }
 
-// ClusterComputeNodeGroup is a compute node group defined in a Cluster
-// resource
-type ClusterComputeNodeGroup struct {
-	ClusterNodeGroup
+// ClusterStatus contains the status for a cluster
+type ClusterStatus struct {
+	// MachineSetCount is the number of actual machine sets that are active for the cluster
+	MachineSetCount int
 
-	Name string
+	// MasterMachineSetName is the name of the master machine set
+	MasterMachineSetName string
+
+	// InfraMachineSetName is the name of the infra machine set
+	InfraMachineSetName string
+
+	// Secret to access the cluster master when running
```

Comment:
Could we rename or comment this to indicate it's a bootstrap concept for nodes to join the cluster.

Reply:
It's not only for bootstrapping, but for general access to the cluster by the cluster operator.

Reply:
Will update the comment